torch/fx/experimental/symbolic_shapes.py

```diff
@@ -269,20 +326,68 @@ def eval(cls, base, divisor):
     'sym_sqrt': lambda a: sympy.sqrt(a),
 }
 
+sizes_strides_methods = {
+    'is_non_overlapping_and_dense': lambda *args: IsNonOverlappingAndDenseIndicator(*args),
+}
+
+def eval_is_non_overlapping_and_dense(sizes, strides):
+    dim = len(sizes)
+
+    # Short-circuits for tensors of rank one, which are
+    # non-overlapping and "dense" if their stride is one
+    if dim == 1:
+        return strides[0] == 1
```

Comment:
Isn't this also true if sizes[0] < 2 ?

Reply:
This is a direct copy from the existing code. I suppose it's still sound even if the short circuit does not work out.